### PR TITLE
Gc/cleanup

### LINKED ISSUE #3: Expand terminal size to fit window
Currently, the terminal in the `FSLeyes GUI` has a fixed size. It would be nice if we could change this to expand when we expand the window.

### LINKED ISSUE #7: Terminal doesn't show by default
The GUI in FSLeyes doesn't show the terminal by default, due to the default placement being in between the other two panels (there isn't enough space). If you drag the panel to a different position, then the terminal will show up. It would be nice to have the terminal displayed by default. However, with FSLeyes/wxPython, this isn't a trivial task, so this issue may require some time to fix (if it is possible).

### PR DESCRIPTION
**Pull request by gaspardcereza**
## Description
General cleanup/improvement of the plugin:
- Remove useless functions/classes
- Move ST window to top
- Move terminal in separate window and use the same with all tabs
- Resize scroll window virtual size with content when elements are added
- Add logos/icons for aesthetic purpose

#### Other potential ideas:
- In #4, Ainsleigh suggested to place the run button on top so that it's always visible. This makes sense but I did'nt know where to put it. 
- I could be good to move the info sizer (on the left) out of the scrolling window but I haven't figured that out yet.
-Separate the code into different files to make it easier to navigate and avoid a file with 2000+ lines. 
 
## Linked Issues
Closes #3 
Closes #7

### CODE REVIEW

File: fsleyes_plugin_shimming_toolbox/st_plugin.py [hunk 1]
```diff
@@ -48,163 +48,110 @@
...
+        self.sizer.SetMinSize((600, 400))
+        self.SetSizer(self.sizer)
 
 
-class Tab(wx.Panel):
+class Tab(wx.ScrolledWindow):
     def __init__(self, parent, title, description):
-        wx.Panel.__init__(self, parent)
+        super().__init__(parent)
+        self.teststring="TEST"
```
Comment (po09i) [inline fsleyes_plugin_shimming_toolbox/st_plugin.py]: Debug?

File: fsleyes_plugin_shimming_toolbox/st_plugin.py [hunk 2]
```diff
@@ -48,163 +48,110 @@
...
     def defaultLayout():
-        """This method makes the control panel appear on the bottom of the FSLeyes window."""
-        return {
-            "location": wx.BOTTOM,
-            "title": "Shimming Toolbox"
-        }
+        """This method makes the control panel appear on the top of the FSLeyes window."""
+        return { "location": wx.TOP, "title": "Shimming Toolbox"}
 
+    def __init__(self, parent, overlayList, displayCtx, ctrlPanel):
```
Comment (po09i) [inline fsleyes_plugin_shimming_toolbox/st_plugin.py:77]: __init__ should be at the very top

File: fsleyes_plugin_shimming_toolbox/st_plugin.py [hunk 3]
```diff
@@ -48,163 +48,110 @@
...
 
     @staticmethod
     def defaultLayout():
-        """This method makes the control panel appear on the bottom of the FSLeyes window."""
-        return {
-            "location": wx.BOTTOM,
-            "title": "Shimming Toolbox"
-        }
+        """This method makes the control panel appear on the top of the FSLeyes window."""
+        return { "location": wx.TOP, "title": "Shimming Toolbox"}
```
Comment (po09i) [inline fsleyes_plugin_shimming_toolbox/st_plugin.py]:
```suggestion
        return {"location": wx.TOP, "title": "Shimming Toolbox"}
```

File: fsleyes_plugin_shimming_toolbox/st_plugin.py [hunk 4]
```diff
@@ -214,59 +161,42 @@ def __init__(self, panel, list_components=[]):
     def create_sizer(self):
         raise NotImplementedError
 
+    # make sure that the create_sizer method has been implemented in the subclasses
     @classmethod
     def __subclasshook__(cls, subclass):
-        return (hasattr(subclass, 'create_sizer') and
-                callable(subclass.create_sizer) or
-                NotImplemented)
+        return (hasattr(subclass, 'create_sizer') and callable(subclass.create_sizer) or NotImplemented)
```
Comment (po09i) [inline fsleyes_plugin_shimming_toolbox/st_plugin.py]: Remove not necessary parenthesis.

File: installer/install_shimming_toolbox.sh [hunk 5]
```diff
@@ -37,4 +37,4 @@ cp "${ST_DIR}/shimming-toolbox/config/coil_config.json" "${ST_DIR}/coil_config.j
 $ST_DIR/python/envs/pst_venv/bin/python -m pip install -e . --quiet
 
 print info "To launch the plugin, load the environment variables then run:"
-print list "shimming-toolbox"
+print list "shimming-toolbox"
```
Comment (po09i) [inline installer/install_shimming_toolbox.sh]: Add blank line

**Review by po09i — COMMENTED** [review]
Suggestion: Add a brief doctoring description for undocumented classes.

I took a brief look at the PR

**Comment by po09i** [comment]
The button looks a bit weird: Screen Shot 2022-04-12 at 11 21 44 AM

File: fsleyes_plugin_shimming_toolbox/st_plugin.py (hunk 1 above)
Comment (gaspardcereza) [reply fsleyes_plugin_shimming_toolbox/st_plugin.py]: Indeed 😅

File: fsleyes_plugin_shimming_toolbox/st_plugin.py (hunk 2 above)
Comment (gaspardcereza) [reply fsleyes_plugin_shimming_toolbox/st_plugin.py:77]: That's how they do it in fsleyes: https://github.com/pauldmccarthy/fsleyes/blob/master/fsleyes/plugins/controls/overlayinfopanel.py

**Comment by gaspardcereza** [comment]
> The button looks a bit weird: Screen Shot 2022-04-12 at 11 21 44 AM

That's weird it looks ok on my end: 
image

File: fsleyes_plugin_shimming_toolbox/st_plugin.py [hunk 6]
```diff
@@ -48,163 +48,109 @@
...
-        super().__init__(parent=parent)
+        Generates the widgets and adds them to the panel.
 
+        """
+        super().__init__(parent, overlayList, displayCtx, ctrlPanel)
+        # Create a notebook to navigate between the different functions.
         nb = wx.Notebook(self)
+
+        # Add a terminal to log messages
+        nb.terminal_component = Terminal(self)
```
Comment (po09i) [inline fsleyes_plugin_shimming_toolbox/st_plugin.py]: The notebook does not have a `terminal_component`, we should wrap the notebook in a class if we want to add an attribute to it. I'll implement a version in the next commit

File: fsleyes_plugin_shimming_toolbox/st_plugin.py [hunk 7]
```diff
@@ -464,8 +393,9 @@ def create_sizer(self):
 
     def add_button_run(self):
         """Add the run button which will call the ``Shimming Toolbox`` CLI."""
-        button_run = wx.Button(self.panel, -1, label="Run")
+        button_run = wx.Button(self.panel, -1, label="", size=(48, 48))
         button_run.Bind(wx.EVT_BUTTON, self.button_run_on_click)
+        button_run.SetBitmap(play_icon, dir=wx.TOP)
```
Comment (po09i) [inline fsleyes_plugin_shimming_toolbox/st_plugin.py]: This is most likely the problem for the run button being too high

File: fsleyes_plugin_shimming_toolbox/st_plugin.py (hunk 6 above)
Comment (po09i) [reply fsleyes_plugin_shimming_toolbox/st_plugin.py]: Added here: https://github.com/shimming-toolbox/fsleyes-plugin-shimming-toolbox/pull/40/commits/4631321f4567bb8d9ae85303c28252034c49f241

File: fsleyes_plugin_shimming_toolbox/st_plugin.py (hunk 7 above)
Comment (po09i) [reply fsleyes_plugin_shimming_toolbox/st_plugin.py]: Added here: https://github.com/shimming-toolbox/fsleyes-plugin-shimming-toolbox/pull/40/commits/2cbc43ecfd2596198058bfdc461819c966abb8c9

**Comment by po09i** [comment]
@gaspardcereza I reviewed it a second time, let me know what you think of the changes!

**Review by po09i — APPROVED** [review]

**Pull request merged**